quinn-h3/examples/h3.rs

```diff
@@ -449,47 +460,93 @@ impl ControlStream {
 
 enum RecvRequestState {
     Receiving(FrameStream<RecvStream>, SendStream),
+    Ready,
 }
 
 struct RecvRequest {
-    state: Option<RecvRequestState>,
+    state: RecvRequestState,
     conn: H3ConnectionRef,
+    streams: Option<(FrameStream<RecvStream>, SendStream)>,
+    headers: Option<HeadersFrame>,
 }
 
 impl RecvRequest {
     fn new(recv: RecvStream, send: SendStream, conn: H3ConnectionRef) -> Self {
         Self {
             conn,
-            state: Some(RecvRequestState::Receiving(FrameStream::with(recv), send)),
+            state: RecvRequestState::Receiving(FrameStream::with(recv), send),
+            streams: None,
+            headers: None,
         }
     }
 }
 
 impl Future for RecvRequest {
-    type Item = RecvRequest;
+    type Item = RequestReady;
     type Error = Error;
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-        println!("RecvRequest polled");
-        // let state = if let Some(state) = self.state {
-        //     std::mem::replace(&mut self.state, None)
-        // };
-        // match state {
-        //     RecvRequestState::Receiving(frames, send) => {
-        //         let frames = try_ready!(frames.poll());
-
-        //     }
-        // }
+        let new_state = match &mut self.state {
+            RecvRequestState::Receiving(frames, _) => {
+                let frame = try_ready!(frames.poll());
+                match frame {
+                    None => return Err(format_err!("recieved an empty request")),
+                    Some(HttpFrame::Headers(f)) => {
+                        self.headers = Some(f);
+                        RecvRequestState::Ready
```

Comment:
You can probably do the replacement in here?

Reply:
Do you mean using `mem::replace` ?

Reply:
Yes.